### PR TITLE
Log warning in rest client response handler 

### PR DESCRIPTION
**Pull request by dreamer-89**
Signed-off-by: Suraj Singh <surajrider@gmail.com>

### Description
Log warnings in rest client response handler causing test failures. This change is to get insights on warnings which is causing the test failures in gradle check.  

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
start gradle check

**Review by nknize — APPROVED** [review]
LGTM, thx!

**Comment by nknize** [comment]
@dblock it's going to fail because of #2334

**Comment by nknize** [comment]
Ignore me... still failing w/ fix

**Comment by dblock** [comment]
I think we need to rebase this PR to pickup that change and run the check again. @dreamer-89 Care to try?

**Comment by nknize** [comment]
Let's wait until #2342 is merged... `TranslogPolicyIT` should no longer run on 2.0.

**Comment by nknize** [comment]
> Will wait for #2342 to merge before rebasing

ready for rebase

File: client/rest/src/main/java/org/opensearch/client/RestClient.java
Comment (andrross) [inline client/rest/src/main/java/org/opensearch/client/RestClient.java:346]: log-and-throw is generally an anti-pattern. We really can't know whether logging is the right thing to do in all contexts for a general purpose client like this, so it is usually better to let whatever handles the exception make the logging decision. In the context of the tests it is definitely a good idea to log these details. Is it possible to handle this exception in the test runner somewhere and log it there?

Comment (dblock) [reply client/rest/src/main/java/org/opensearch/client/RestClient.java:346]: I agree with @andrross here, wdyt @dreamer-89?

**Review by dblock — CHANGES_REQUESTED** [review]
I went too fast doing a CR, I think I agree with Andrew.

File: client/rest/src/main/java/org/opensearch/client/RestClient.java
Comment (dreamer-89) [reply client/rest/src/main/java/org/opensearch/client/RestClient.java:346]: Thanks @andrross @dblock for the feedback. Yes, I also agree with @andrross here. 

The test `recoverReplica` directly calls the rest client via parent class `OpenSearchRestTestCase` method. So, there are couple of options for adding log statement;
1.  Inside parent class `OpenSearchRestTestCase` 
2. Individual test cases.

#1 seems better option as its minor change and covers all cases.

I think since issue the original issue is fixed in https://github.com/opensearch-project/OpenSearch/issues/2334; we can hold onto this change for now ?

**Comment by andrross** [comment]
@dreamer-89 Can we close out this PR for now?

**Comment by dreamer-89** [comment]
> @dreamer-89 Can we close out this PR for now?

Sure @andrross . Closing this PR as it is not necessary now. The original issue which we wanted to debug is already fixed in https://github.com/opensearch-project/OpenSearch/issues/2334

**Pull request closed without merging**
